examples/file-transfer/README.md

```diff
@@ -0,0 +1,47 @@
+# File Transfer
+
+This example contains minimal HTTP client and server.
+
+The client uploads a file to the server in a single POST, shaping traffic to send small data chunks.
+
+The server manually processes requests in order to be able to write as soon as data arrives, to avoid buffering a whole (possibly huge) file not fitting in RAM.
```

Comment:
Please mention authentication, e.g. "Only an authenticated users can perform an upload."

Reply:
Two paragraphs below that